# Morning lab

In [1]:
#import statsmodels.api as sm
#from statsmodels.formula.api import ols
import pandas as pd
import numpy as np
import scipy.stats as stats

In [3]:
# •	State the null hypothesis
# Different plasma beam power does not influence the etching rate.

# •	State the alternate hypothesis
# There is at least one plasma beam power that influence the etching rate.

#•	What is the significance level
#alpha = 0.05


#•	What are the degrees of freedom of model, error terms, and total DoF
#Degrees of freedom in numerator – 2
#Degrees of freedom in denominator – 12


In [2]:
data = pd.read_excel('anova_lab_data.xlsx', sheet_name='data_collected')
data

Power   Etching Rate
0   160 W          5.43
1   180 W          6.24
2   200 W          8.79
3   160 W          5.71
4   180 W          6.71
5   200 W          9.20
6   160 W          6.22
7   180 W          5.98
8   200 W          7.90
9   160 W          6.01
10  180 W          5.66
11  200 W          8.15
12  160 W          5.59
13  180 W          6.60
14  200 W          7.55

In [6]:
data.columns = ['Power', 'Etching Rate']

In [7]:
# Global mean:
gm = np.mean(data['Etching Rate'])
print("Global mean",gm)

# Mean of every group:
group_averages = [(group,np.mean(data[data['Power'] == group]['Etching Rate'])) for group in list(data['Power'].unique())] 
print("The group means are: ",group_averages)

Global mean 6.782666666666667
The group means are:  [('160 W', 5.792), ('180 W', 6.2379999999999995), ('200 W', 8.318)]


Now we can compute the numerator of F:

In [8]:
numerator = 0
for group in group_averages:
    numerator += len(data[data['Power'] == group[0]]['Etching Rate'].values) * ( group[1] - gm )**2
numerator = numerator / ( len(data['Power'].unique()) - 1 ) 

print("\hat{S}_{t}^{2} is: ", numerator)

\hat{S}_{t}^{2} is:  9.088326666666667


Now the denominator:

In [9]:
denominator = 0
for group in group_averages: 
    for value in data[data['Power'] == group[0]]['Etching Rate'].values:
        denominator += (value - group[1])**2
denominator = denominator / ( len(data) - len(data['Power'].unique()) )

print("\hat{S}_{e}^{2} is: ",denominator)

\hat{S}_{e}^{2} is:  0.24643666666666644


Now we can compute the ratio:

In [12]:
F = numerator / denominator

print("The F statistic is: ",round(F, 2))

The F statistic is:  36.88


Now let's get the critical value for F. 

Assuming a confidence interval of 95%, our significance level $\alpha$ = 0.05. The degrees of freedom in the numerator is: (number of groups - 1) = 3 - 1 = 2. The number of degrees of freedom for the denominator is: ( number of records - number of groups ) = 15 - 4 = 12. 

If we go to the [F-Table](http://www.socr.ucla.edu/Applets.dir/F_Table.html), we get that the critical value for F is: 2.80680


Now taking the decission using the F-critical value:

As our F-statistic (36.88) > F-ctitical value (3.80), **we reject the H0**